compiler/parser.nim

```diff
@@ -840,11 +855,15 @@ proc primarySuffix(p: var Parser, r: PNode,
       # `foo ref` or `foo ptr`. Unfortunately, these two are also
       # used as infix operators for the memory regions feature and
       # the current parsing rules don't play well here.
-      if p.inPragma == 0 and (isUnary(p.tok) or p.tok.tokType notin {tkOpr, tkDotDot}):
-        # actually parsing {.push hints:off.} as {.push(hints:off).} is a sweet
-        # solution, but pragmas.nim can't handle that
-        result = commandExpr(p, result, mode)
-      break
+      if p.tok.isDotLike: # synchronize with `tkDot` branch
```

Comment:
The grammar does not reflect this rule. You need to patch the `#| primarySuffix = ...` comment and run "nim c -r compiler/parser.nim".

Reply:
done